examples/compress/compress.cpp

```diff
@@ -0,0 +1,645 @@
+#include "arg.h"
+#include "common.h"
+#include "sampling.h"
+#include "sampling.cpp"
```

Comment:
Should not be necessary to include `sampling.cpp`.

Reply:
I can't access directly `gpt_sampler.set_logits()` without it, as it is not exposed in sampling.h. I will try achieving the same thing without it.

Reply:
I figured it out